packages/content-management/content_insights/content_insights_public/src/open_insights_flyout.tsx

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+import React, { useCallback, useRef } from 'react';
+import type { OverlayRef } from '@kbn/core-mount-utils-browser';
+
+import { useServices, ContentInsightsProvider } from './services';
+
+import { InsightsFlyoutContent, InsightsFlyoutContentProps } from './components';
+
+export type OpenInsightsFlyoutParams = Pick<InsightsFlyoutContentProps, 'item' | 'entityName'>;
+
+export function useOpenInsightsFlyout() {
+  const services = useServices();
+  const flyout = useRef<OverlayRef | null>(null);
+
+  return useCallback(
+    ({ ...flyoutProps }: OpenInsightsFlyoutParams) => {
+      if (!services?.openFlyout) {
+        return;
```

Comment:
Instead of being silent about missing services, should we throw a custom error here if the `services` value is null? Otherwise developers do not get strong feedback if they try to call this hook but forgot to put `ContentInsightsProvider` in their rendering tree.

Reply:
agree, changed